libflux/src/flux/benches/everything.flux

```diff
@@ -0,0 +1,5000 @@
+#group,false,false,true,true,false,true,true,false,false
+#default,_result,,,,,,,,
+,result,table,_start,_stop,_time,_measurement,tag0,value_a,value_b
+,,0,2018-12-19T00:00:00Z,2018-12-20T00:00:00Z,2018-12-19T22:13:30Z,_m,g,2,1
+,,0,2018-12-19T00:00:00Z,2018-12-20T00:00:00Z,2018-12-19T22:13:40Z,_m,g,3,2
+,,0,2018-12-19T00:00:00Z,2018-12-20T00:00:00Z,2018-12-19T22:13:50Z,_m,g,4,3
+,,0,2018-12-19T00:00:00Z,2018-12-20T00:00:00Z,2018-12-19T22:14:00Z,_m,g,5,4
+,,0,2018-12-19T00:00:00Z,2018-12-20T00:00:00Z,2018-12-19T22:14:10Z,_m,g,6,5
+,,0,2018-12-19T00:00:00Z,2018-12-20T00:00:00Z,2018-12-19T22:14:20Z,_m,g,7,6
+"
+
+join_test_fn = (table=<-) => {
+    a = table
+        |> range(start: 2018-12-19T00:00:00Z, stop: 2018-12-20T00:00:00Z)
+        |> filter(fn: (r) => r._field == "a")
+        |> drop(columns: ["_field"])
+        |> rename(columns: {_value: "value_a"})
+
+    b = table
+        |> range(start: 2018-12-19T00:00:00Z, stop: 2018-12-20T00:00:00Z)
+        |> filter(fn: (r) => r._field == "b")
+        |> drop(columns: ["_field"])
+        |> rename(columns: {_value: "value_b"})
+
+    return experimental.join(left:a, right:b, fn:(left, right) => ({left with value_b: right.value_b}))
+}
+
+test experimental_join = () =>
+	({input: testing.loadStorage(csv: inData), want: testing.loadMem(csv: outData), fn: join_test_fn})
+package bigtable
+
+builtin from : (token: string, project: string, instance: string, table: string) => [T] where T: Record
+
+package array_test
+
+import "testing"
+import "experimental/array"
+
+option now = () => 2018-12-19T22:14:30Z
+
+data = "
+#datatype,string,long,string,string,string,dateTime:RFC3339,boolean
+#group,false,false,false,false,false,false,false
+#default,_result,,,,,,
+,result,table,_measurement,_field,t0,_time,_value
+,,0,m0,f0,tagvalue,2018-12-19T22:13:30Z,false
+,,0,m0,f0,tagvalue,2018-12-19T22:13:40Z,true
+,,0,m0,f0,tagvalue,2018-12-19T22:13:50Z,false
+,,0,m0,f0,tagvalue,2018-12-19T22:14:00Z,false
+,,0,m0,f0,tagvalue,2018-12-19T22:14:10Z,true
+,,0,m0,f0,tagvalue,2018-12-19T22:14:20Z,true
+"
+
+input = array.from(rows: [{
+	_measurement: "m0",
+	_field: "f0",
+	t0: "tagvalue",
+	_time: 2018-12-19T22:13:30Z,
+	_value: false,
+}, {
+	_measurement: "m0",
+	_field: "f0",
+	t0: "tagvalue",
+	_time: 2018-12-19T22:13:40Z,
+	_value: true,
+}, {
+	_measurement: "m0",
+	_field: "f0",
+	t0: "tagvalue",
+	_time: 2018-12-19T22:13:50Z,
+	_value: false,
+}, {
+	_measurement: "m0",
+	_field: "f0",
+	t0: "tagvalue",
+	_time: 2018-12-19T22:14:00Z,
+	_value: false,
+}, {
+	_measurement: "m0",
+	_field: "f0",
+	t0: "tagvalue",
+	_time: 2018-12-19T22:14:10Z,
+	_value: true,
+}, {
+	_measurement: "m0",
+	_field: "f0",
+	t0: "tagvalue",
+	_time: 2018-12-19T22:14:20Z,
+	_value: true,
+}])
+
+pass = (tables=<-) => tables
+
+test from = () => ({
+	input: input,
+	want: testing.loadMem(csv: data),
+	fn: pass,
+})
+package array_test
```

Comment:
do these repeated package statements parse alright? Or they trigger an error? Again I only care if it means that only part of the file is actually being parsed.

Reply:
They produce errors in my editor and on the command line, but I can't be sure whether that's the failing in the semantic pass or the straight parsing pass. It was a thing I was going to ask for insight on. My understanding was that parsing was only checking tokens.

Reply:
Also, import statements and option statements also cause issues. It's _possible_ that the idea of `cat`ing all the files together isn't actually going to work.

Reply:
Options should be able go anywhere I think. They are in statement. I did this once before and I think I just used `grep -v` for the `package` and `options` statements, and also `sed` to take out the strings from the test cases so it didn't become a string heavy benchmark. That was before I had a colm grammar for flux. If you're interested in trying out a source code analysis tool, I can show you how to do it in colm. The flux grammar is fairly complete and it's pretty quick to pattern match and take out statements.

Reply:
It looks like the entire file is parsed, with error nodes added where needed.

Reply:
I've pared down the file to be completely valid flux. It parses correctly, with no errors, just to simplify things. In the process of doing this and experimenting, it did become clear that there is indeed much low hanging fruit in parsing performance (particularly around parsing errors).

Reply:
Right, I guess you could include invalid flux as the parser recovers and and pushes on. The original goal was to look for regressions in scanning, so in that case my thoughts are to stick to valid flux to eliminate the possibility that parsing gets aborted. But in general it does make a lot of sense to benchmark error recovery.

Reply:
I agree. At some point we're going to come back to performance, and we can expand those benchmarks (and add them to CI) as we do that work.